GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -435,6 +435,38 @@ protected override void OnHandleCreated(EventArgs e)
             base.OnHandleCreated(e);
         }
 
+        public void FixForOwnerScreenLocation(IWin32Window? owner)
+        {
+            if (owner != null)
+            {
+                Screen screenOwner = Screen.FromHandle(owner.Handle);
+                if (!screenOwner.Bounds.IntersectsWith(Bounds))
+                {
+                	// Fix #10121 - Commit form appear out of screen in dual monitor setup
+                	//
+                    // This form appear to be not in the same monitor as owner window
+                    // thus may become invisible when owner is in secondary screen
+                    // with negative left coordinates and last saved position of the
+                    // form was with positive coordinates in case dpi are different.
+                    // Attempting to correct the saved position may be unsuccessful
+                    // because it depends on the screen layout of extended desktop.
+                    if (screenOwner.Bounds.Width - screenOwner.Bounds.X < 0 && Left > 0)
+                    {
+                        // Move to the owner screen
+                        Left = screenOwner.Bounds.X + Left;
+
+                        // Show in taskbar so user can potentially recover the window
+                        // if the title bar is no more reachable by user.
+                        ShowInTaskbar = true;
```

Comment:
We mustn't change this property. It performs [non-trivial amount of work](https://github.com/dotnet/winforms/blob/01abac5b35616f182267222da1976d429e1eab19/src/System.Windows.Forms/src/System/Windows/Forms/Form.cs#L1938-L1941), and creates an inconsistent UX.